cvmfs/mountpoint.cc

```diff
@@ -510,56 +567,67 @@ bool FileSystem::SetupNfsMaps() {
 }
 
 
-bool FileSystem::SetupQuotaMgmt() {
-  assert(quota_limit_ >= 0);
-  int64_t quota_threshold = quota_limit_ / 2;
-  PosixQuotaManager *quota_mgr;
-
-  if (cache_mode_ & FileSystem::kCacheShared) {
-    quota_mgr = PosixQuotaManager::CreateShared(
-                  exe_path_,
-                  cache_dir_,
-                  quota_limit_,
-                  quota_threshold,
-                  foreground_);
-    if (quota_mgr == NULL) {
-      boot_error_ = "Failed to initialize shared lru cache";
-      boot_status_ = loader::kFailQuota;
-      return false;
-    }
-  } else {
-    // Cache database should to be protected by workspace lock
-    assert(workspace_ == cache_dir_);
-    quota_mgr = PosixQuotaManager::Create(
-                  cache_dir_,
-                  quota_limit_,
-                  quota_threshold,
-                  found_previous_crash_);
-    if (quota_mgr == NULL) {
-      boot_error_ = "Failed to initialize lru cache";
-      boot_status_ = loader::kFailQuota;
-      return false;
+bool FileSystem::SetupQuotaMgmt(cache::CacheManagerIds mgr_type) {
```

Comment:
Probably easier to start with an `if {` block for the RAM cache manager with an early return.  The rest of the code can then stay unmodified.